File: code/ch09/9.1.1.2.two_sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        for (int left = 0, right = numbers.size() - 1; left < right; ){
            if (numbers[left] + numbers[right] == target)
                return vector<int>{left + 1, right + 1};
            else if (numbers[left] + numbers[right] < target)
                left++;
            else
                right--;
        }
        return vector<int>{};
    }
};

File: code/ch09/9.1.1.2.two_sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int left = 0, right = nums.length - 1;
        while (left < right) {
            if (nums[left] + nums[right] == target)
                return new int[]{left + 1, right + 1};
            else if (nums[left] + nums[right] < target)
                left++;
            else
                right--;
        }
        return null;
    }
}